File: RACDemo/RACDemo/ViewController.m
//
//  ViewController.m
//  RACDemo
//
//  Created by iMac on 2017/11/23.
//  Copyright © 2017年 iMac. All rights reserved.
//

#import "ViewController.h"
#import "BaseViewController.h"

@interface ViewController ()<UITableViewDelegate,UITableViewDataSource>

@property (weak, nonatomic) IBOutlet UITableView *tableView;
@property(nonatomic, strong) NSArray *viewControllers;
@property(nonatomic, strong) NSArray *titles;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.viewControllers.count;
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    static NSString *identifier = @"cell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:identifier];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:identifier];
    }
    cell.textLabel.text = self.titles[indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    UIStoryboard *story = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    NSString *identifier = self.viewControllers[indexPath.row];
    BaseViewController *baseVc = [story instantiateViewControllerWithIdentifier:identifier];
    [self.navigationController pushViewController:baseVc animated:YES];
}


#pragma mark - setter/getter

- (NSArray *)viewControllers{
    if(_viewControllers == nil){
        _viewControllers = @[@"RACSignalViewController",
                             @"RACSubjectViewController",
                             @"RACCommandViewController",
                             @"RACSequenceViewController",
                             @"RACDefineViewController",
                             @"RACBindViewController",
                             @"RACMapViewController",
                             @"RACIgnoreViewController",
                             @"RACConcatViewController",
                             @"RACMulticastConnectionVC",
                             @"NetWorkViewController"
                             ];
    }
    return _viewControllers;
}
- (NSArray *)titles{
    if (_titles == nil) {
        _titles = @[@"Signal",
                    @"Subject",
                    @"Command",
                    @"Sequence",
                    @"常用宏",
                    @"Bind",
                    @"映射",
                    @"过滤",
                    @"组合",
                    @"RACMulticastConnectionVC",
                    @"网络请求"
                    ];
    }
    return _titles;
}




- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: SwiftKit/SwiftKit/ViewController.swift
//
//  ViewController.swift
//  SwiftKit
//
//  Created by iMac on 2017/11/10.
//  Copyright © 2017年 iMac. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

/*
extension ViewController:UITableViewDelegate,UITableViewDataSource{
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 4
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell:UITableViewCell = UITableViewCell(style:UITableViewCellStyle.subtitle, reuseIdentifier: "cells")
        cell.textLabel?.text = "textLabel"
        cell.detailTextLabel?.text = "detailText"
        cell.imageView?.image = UIImage(named:"123")
        return cell
    }
    
    func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
        return "Footer"
    }
    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
        return "Header"
    }
    
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 80
    }
    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
        return 40
    }
    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
        return 40
    }
}
*/
